### PR TITLE
Issue #167 - update Adapter docs

### PR DESCRIPTION
**Pull request by Skejven**
Adapters documentation

---

I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: documentation/src/main/wiki/Adapter.md [hunk 1]
```diff
@@ -1,7 +1,59 @@
...
+- [[Service Adapter|ServiceAdapter]] for [[Action Knot|ActionKnot]],
+- [[Action Adapter|ActionAdapter]] for [[View Knot|ViewKnot]].
+
+Knot.x comes with implementation of a [[Service Adapter|ServiceAdapter]], that enables communication with external 
+services using HTTP Protocol. See [[Http Service Adapter|HttpServiceAdapter]] for more information.
+Please note, that this implementation is very generic and we recommend to create project-specific 
+adapters for any custom solution.
+
+## How to configure?
+Adapter may have its configuration in form of JSON object entry.
```
Comment (tomaszmichalak) [inline documentation/src/main/wiki/Adapter.md]: may have -> has

Comment (tomaszmichalak) [inline documentation/src/main/wiki/Adapter.md]: Maybe we should provide here information about generic adapter configuration model: com.cognifide.knotx.adapter.api.AdapterConfiguration.

File: documentation/src/main/wiki/Adapter.md [hunk 2]
```diff
@@ -1,7 +1,59 @@
...
+adapters for any custom solution.
+
+## How to configure?
+Adapter may have its configuration in form of JSON object entry. 
+Please see example configuration for [[Http Service Adapter|HttpServiceAdapter#how-to-configure]]
+
+## How to extend?
+| ! Note |
+|:------ |
+| Please note that this section explains how to write custom adapter using Java. But this is not the only way to connect with Knot.x thanks to [Vert.x polyglotism](http://vertx.io/). |
```
Comment (tomaszmichalak) [inline documentation/src/main/wiki/Adapter.md]: But this is not the only way to connect with Knot.x thanks to [Vert.x polyglotism](http://vertx.io/).
->
Thanks to [Vert.x polyglotism mechanism](http://vertx.io/) you can implement your Adapters in language you like.

File: documentation/src/main/wiki/HttpServiceAdapter.md [hunk 3]
```diff
@@ -1 +1,213 @@
...
+- `params` - JSON object that contains additional parameters, among those parameter mandatory `path` parameter should be defined.
+
+### Service path
+`path` parameter is a mandatory parameter that must be passed to Http Service Adapter. 
+It defines request path and may contain [placeholders](#parametrized-service-calls).
+
+### Parametrized services calls
+When found a placeholder within the `path` parameter it will be replaced with a dynamic value based on the current http request (data from `clientRequest`).
+Available placeholders are:
+* `{header.x}` - is the original requests header value where `x` is the header name
```
Comment (tomaszmichalak) [inline documentation/src/main/wiki/HttpServiceAdapter.md]: original -> client

File: documentation/src/main/wiki/HttpServiceAdapter.md [hunk 4]
```diff
@@ -1 +1,213 @@
...
+
+- Space character is substituted by `%20` instead of `+`.
+- Slash character `/` remains as it is.
+
+### Adapter Response
+Http Service Adapter replies with `ClientResponse` that contains:
+
+| Parameter       | Type                      |  Description  |
+|-------:         |:-------:                  |-------|
+| `statusCode`    | `Number`                  | status code of a response from external service (e.g. `200`) |
```
Comment (tomaszmichalak) [inline documentation/src/main/wiki/HttpServiceAdapter.md]: Number -> HttpResponseStatus

File: documentation/src/main/wiki/HttpServiceAdapter.md [hunk 5]
```diff
@@ -1 +1,213 @@
...
+| `headers`       | `MultiMap`                | external service response headers |
+| `body`          | `Buffer`                  | external service response, **please notice that it is expected, tha form of a response body from an external service is JSON** |
+
+## How to configure?
+Http Service Adapter can be configured to support multiple external `services`, see example configuration:
```
Comment (tomaszmichalak) [inline documentation/src/main/wiki/HttpServiceAdapter.md]: Please follow convention from other modules.

File: documentation/src/main/wiki/HttpServiceAdapter.md (hunk 3 above)
Comment (Skejven) [inline documentation/src/main/wiki/HttpServiceAdapter.md]: Fixed

File: documentation/src/main/wiki/HttpServiceAdapter.md (hunk 5 above)
Comment (Skejven) [inline documentation/src/main/wiki/HttpServiceAdapter.md]: Fixed.

**Pull request merged**